app/code/Magento/Sales/Model/Increment.php

```diff
@@ -51,8 +66,13 @@ public function getCurrentValue()
      */
     public function getNextValue($storeId)
     {
+		if(empty($this->entityType)){
+			throw new \Magento\Framework\Model\Exception(
+				__('EntityType is empty')
```

Comment:
As @airbone42 pointed out, if you want this to be a required argument then it should be injected in the constructor instead of by the `setEntityType` method. 

    public function __construct(
         EavConfig $eavConfig,
         $entityType = Order::ENTITY
     ) {
         $this->eavConfig = $eavConfig;
         $this->entityType = $entityType;
     }

Please also look at the failing tests and add new test cases for injecting non-default entity types.